pkg/model/printer_column.go

```diff
@@ -0,0 +1,144 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package model
+
+import (
+	"fmt"
+	"sort"
+)
+
+// PrinterColumn represents a single field in the CRD's Spec or Status objects
+type PrinterColumn struct {
+	CRD      *CRD
+	Name     string
+	Type     string
+	JSONPath string
+}
+
+// By can sort two PrinterColumns
+type By func(a, b *PrinterColumn) bool
+
+// Sort does an in-place sort of the supplied printer columns
+func (by By) Sort(subject []*PrinterColumn) {
+	pcs := printerColumnSorter{
+		cols: subject,
+		by:   by,
+	}
+	sort.Sort(pcs)
+}
+
+// printerColumnSorter sorts printer columns by name
+type printerColumnSorter struct {
+	cols []*PrinterColumn
+	by   By
+}
+
+// Len implements sort.Interface.Len
+func (pcs printerColumnSorter) Len() int {
+	return len(pcs.cols)
+}
+
+// Swap implements sort.Interface.Swap
+func (pcs printerColumnSorter) Swap(i, j int) {
+	pcs.cols[i], pcs.cols[j] = pcs.cols[j], pcs.cols[i]
+}
+
+// Less implements sort.Interface.Less
+func (pcs printerColumnSorter) Less(i, j int) bool {
+	return pcs.by(pcs.cols[i], pcs.cols[j])
+}
+
+// AdditionalPrinterColumns returns a sorted list of PrinterColumn structs for
+// the resource
+func (r CRD) AdditionalPrinterColumns() []*PrinterColumn {
+	byName := func(a, b *PrinterColumn) bool {
+		return a.Name < b.Name
+	}
+	By(byName).Sort(r.additionalPrinterColumns)
+	return r.additionalPrinterColumns
+}
+
+// addPrintableColumn adds an entry to the list of additional printer columns
+// using the given path and field types.
+func (r *CRD) addPrintableColumn(
+	field *Field,
+	jsonPath string,
+) *PrinterColumn {
```

Comment:
nit: Maybe we should stop returning `*PrinterColumn` ? `AddStatusField` and `AddSpecField` don't use it

Reply:
nice callout. done.